fix(editors): use indexOf in multiple select editor to load value

jmzagorski opened the pull request:
`indexOf` must be used because `defaultValue` is an array for the `multipleSelectEditor`. the current `multipleSelectEditor` is broken and values are not loaded when the multiple select list is opened.

This reverts part of this [commit](https://github.com/ghiscoding/aurelia-slickgrid/commit/71916dc8bd80e257c310fa89485fc8a2e534a5d1) for the `multipleSelectEditor`.

The excel export and the sort still works with numbers. However, the current state of sorters will not sort "Task 2" and "Task 10" correctly regardless of the editors because those values are strings and not numbers. A separate issue would have to be opened to that. I updated `Example3.ts` to show that problem by allowing in Tasks greater than 10 in the prerequisite column.

If I can figure out/have permission to release a hot fix (version 1.12.3) I will do that because I would imagine this is critical since it breaks inline editing.

ghiscoding commented:
Hmm it does revert my previous [commit](https://github.com/ghiscoding/aurelia-slickgrid/commit/71916dc8bd80e257c310fa89485fc8a2e534a5d1), have you tested the sort descending which was a fix in that previous commit? I'm a bit surprised since it's a complete revert of my previous commit, which was a fix. Anyhow, I don't use this inline editor in any of my projects yet, so you would more than I do on this subject.

For the string sorting, I would actually rather always sort by string as default. We might need another option to have another way to sort. If another option is created for that, we could take advantages of the built-in [Sorters](https://github.com/ghiscoding/aurelia-slickgrid/tree/master/aurelia-slickgrid/src/aurelia-slickgrid/sorters), these are used to sort data of local grid (without a Backend Service).

jmzagorski commented:
Would I be testing the data sorted on the grid or in the select list?

ghiscoding commented:
The previous commit I made was to fix the select list not being sorted correctly on a sort order descending. Does that answer your question?

jmzagorski commented:
Yes, thank you. I will double check the sorting DESC. Not sure how much work I will be able to do over the next week (i will be "off work", but may still be working lol), but this will be the next thing I will look into.

ghiscoding commented:
@jmzagorski 
Not totally related to this PR, but can you give an update on what has to be reviewed? 
I find there's a lot of open issues and I'd like to close a few, if possible. 

- For the current PR, I guess you will look into it on the short term
- issue #44 for the sort icon now has my PR/fix merged in SlickGrid, however waiting for a new release so that I can then apply the fix on this lib as well
- PR #51, I will take a look in the coming weekend to see if we can get rid of the function arguments all together, else will probably merge what you already submitted
- Issue #48 with Froala editor, that's a though one, not sure about it, that might be hard to fix since it might be a conflict with the jquery drag event lib
- issue #36, multiple grids in 1 view. This interest me a lot and I guess you will work on that one. The question is, will that be doable in version `1.x` or will that be in `2.x` ?

jmzagorski commented:
I reviewed example 3 again for sorting editors and added examples in 12 and 4 for filters. If you pull down the branch I can't find any problems with sorting.  Could I be missing something you saw? The `singleSelectEditor` was definitely wrong and you fixed that in the commit, but maybe the `multipleSelectEditor` was okay?

Also, the `multipleSelectFilter` and `singleSelectFilter` has a check for `this.gridOptions.params` [here](https://github.com/ghiscoding/aurelia-slickgrid/blob/master/aurelia-slickgrid/src/aurelia-slickgrid/filters/multipleSelectFilter.ts#L95) and I am not sure why since that is never used in the sort or filter.

Another thing to be aware of, if there is a DESC sortBy set on a filter, the blank default option will be thrown to the bottom. You can see this in my example 12 I added.

ghiscoding commented:
oh yeah I guess that is expected for a sort DESC lol. Not much we can do about that I think. Unless we fix it with some kind of patch but that is kind of ugly

jmzagorski commented:
I removed the `gridOptions` check, but kept the check on the `columnDef`

ghiscoding commented:
Oh yeah that does look good. Where you done with the PR then, anything else to review before the merge? I'm good with your changes but I didn't test locally and I don't think I need to since you tested yourself.

jmzagorski commented:
i am playing around with all the examples to see if an exceptions are thrown and if anything weird happens. After that I can merge it if you want.

Not related, do you notice the inline date editors not working? I select a date and it does not set it.

ghiscoding commented:
Yup go ahead and merge whenever ready 👍 

There's also this other [PR](https://github.com/ghiscoding/aurelia-slickgrid/pull/54) that you can merge after reviewing. That one, I was going to merge it myself but I know you use Inline Editor, so I left it open for you to be aware of, there's some description in it

jmzagorski commented:
Oh yea I saw that PR and added a review I never submitted last night. I submitted it now
